pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java

```diff
@@ -161,6 +161,10 @@
     public static final double DEFAULT_BROKER_MIN_RESOURCE_PERCENT_FOR_START = 100.0;
     public static final String CONFIG_OF_ENABLE_QUERY_LIMIT_OVERRIDE = "pinot.broker.enable.query.limit.override";
 
+    // Config for number of threads to use for Broker reduce-phase.
+    public static final String CONFIG_OF_NUM_REDUCE_THREADS = "pinot.broker.num.reduce.threads";
+    public static final int DEFAULT_NUM_REDUCE_THREADS = 1; // TBD: Change to a more appropriate default (eg numCores).
```

Comment:
Actually, on second thought, the default value of `1` is not good, as it will make reduce across concurrent queries as sequential. Moreover, if we add more threads, then it may cause contention in case of high qps use cases.

While we tune this, perhaps the `DEFAULT_NUM_THREADS = 1` should fall back to the current behavior without any executor service and any value `> 1` should use the executor service with specified threads.

Thoughts @kishoreg  @Jackie-Jiang ?

Reply:
This config is right but the implementation can be changed. This should be something similar to what we have in combine operator - Executor pool is cached or capped at a high number based on the number of cores. But the number of callables we create be based on this config.